src/genomics/structures.cxx

```diff
@@ -1,31 +1,53 @@
 #include <assert.h>
+#include <tuple>
 
 #include "genomics/structures.hpp"
 
 namespace genomics {
-  coordinates resolve_absolute(const genome_structure& gs, uint64_t absolute_coords) {
-    for (const chromosome& chr : gs) {
-      if (absolute_coords < chr.length) {
-        coordinates c = { chr, absolute_coords };
-        return c;
-      }
-
-      absolute_coords -= chr.length;
-    }
+    std::tuple<coordinates, std::string> resolve_absolute(const genome_structure& gs, int64_t off_target_abs_coords, const kmer& k) {
```

Comment:
All calculations for position/strand are now part of this function to reduce complexity outside it.